internal/provider/resource_access_jaas_group.go

```diff
@@ -0,0 +1,112 @@
+	return &jaasAccessGroupResource{genericJAASAccessResource: genericJAASAccessResource{
+		targetResource:  groupInfo{},
+		resourceLogName: LogResourceJAASAccessGroup,
+	}}
+}
+
+type groupInfo struct{}
+
+// Info implements the [resourceInfo] interface, used to extract the info from a Terraform plan/state.
+func (j groupInfo) Info(ctx context.Context, getter Getter, diag *diag.Diagnostics) (genericJAASAccessData, names.Tag) {
+	groupAccess := jaasAccessModelResourceGroup{}
+	diag.Append(getter.Get(ctx, &groupAccess)...)
+	accessGroup := genericJAASAccessData{
+		ID:              groupAccess.ID,
+		Users:           groupAccess.Users,
+		Groups:          groupAccess.Groups,
+		ServiceAccounts: groupAccess.ServiceAccounts,
+		Access:          groupAccess.Access,
+	}
+	// When importing, the group name will be empty
+	var tag names.Tag
+	if groupAccess.GroupID.ValueString() != "" {
+		tag = jimmnames.NewGroupTag(groupAccess.GroupID.ValueString())
+	}
+	return accessGroup, tag
+}
+
+// Save implements the [resourceInfo] interface, used to save info on Terraform's state.
+func (j groupInfo) Save(ctx context.Context, setter Setter, info genericJAASAccessData, tag names.Tag) diag.Diagnostics {
+	groupAccess := jaasAccessModelResourceGroup{
+		GroupID:         basetypes.NewStringValue(tag.Id()),
+		ID:              info.ID,
+		Users:           info.Users,
+		Groups:          info.Groups,
+		ServiceAccounts: info.ServiceAccounts,
+		Access:          info.Access,
+	}
+	return setter.Set(ctx, groupAccess)
+}
+
+// ImportHint implements [resourceInfo] and provides a hint to users on the import string format.
+func (j groupInfo) ImportHint() string {
+	return "group-<id>:<access-level>"
+}
+
+type jaasAccessGroupResource struct {
+	genericJAASAccessResource
+}
+
+type jaasAccessModelResourceGroup struct {
+	GroupID         types.String `tfsdk:"group_id"`
+	Users           types.Set    `tfsdk:"users"`
+	ServiceAccounts types.Set    `tfsdk:"service_accounts"`
+	Groups          types.Set    `tfsdk:"groups"`
+	Access          types.String `tfsdk:"access"`
+
+	// ID required for imports
+	ID types.String `tfsdk:"id"`
+}
+
+// Metadata returns metadata about the JAAS group access resource.
+func (a *jaasAccessGroupResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_jaas_access_group"
+}
+
+// Schema defines the schema for the JAAS group access resource.
+func (a *jaasAccessGroupResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attributes := a.partialAccessSchema()
+	attributes["group_id"] = schema.StringAttribute{
+		Description: "The ID of the group for access management. If this is changed the resource will be deleted and a new resource will be created.",
+		Required:    true,
+		Validators: []validator.String{
+			ValidatorMatchString(jimmnames.IsValidGroupId, "group must be a valid ID"),
```

Comment:
todo: add description of what a valid group ID is.

Reply:
Made a very small tweak to switch from ID to UUID.